package com.example.phoenix.context.exception.handler;

public enum ErrorCode {
  ALREADY_EXISTS_ERROR_CODE(1100), // ==
  USER_ALREADY_EXISTS(1100),
  BOOK_NOT_FOUND(123), // ==
  USER_PHONE_EXIST(1100), // == // user exist with this number
  NOT_FOUND_ERROR_CODE(1200),
  URL_NOT_FOUND(1201),
  USER_NOT_FOUND(1202),
  ORDER_NOT_FOUND(1203),
  INVALID_CARD_NUMBER(1203),

  INVALID_TOKEN_ERROR_CODE(1700), // ==
  INVALID_OPERATION_ERROR_CODE(1800), // ==
  INVALID_ARGUMENT_ERROR_CODE(1234),
  REQUIRED_FIELD_MISSED(1888), // ==
  UNAUTHORIZED_ERROR_CODE(2000), // ==

  FORBIDDEN_ERROR_CODE(2100), // ==

  USER_BLOCKED_ERROR_CODE(2200), // ==

  RETRIEVE_FORBIDDEN_ERROR_CODE(2600), // ==

  JSON_PARSING_ERROR_CODE(2800), // ==

  TIMEOUT_ERROR_CODE(3200), // ==

  SERVICE_UNAVAILABLE_ERROR_CODE(3300), // ==

  INTERNAL_ERROR_CODE(3400), // ==

  BAD_REQUEST_CODE(3500), // ==
  INVALID_TYPE(3600), // ==

  PASSWORD_NOT_MATCH_ERROR(3700), // ==
  ORDER_ALREADY_PAID(3700), // ==
  IO_EXCEPTION(3600), // ==
  MAX_UPLOAD_SIZE_EXCEEDED(2367); // ==

  public final int value;

  ErrorCode(int value) {
    this.value = value;
  }
}
